malariagen_data/anoph/base.py

```diff
@@ -489,11 +580,6 @@ def _prep_sample_sets_param(
                 # Make sure we end up with a flat list of sample sets.
                 prepped_sample_sets.extend(sp)
 
-        else:
-            raise TypeError(
-                f"Invalid type for sample_sets parameter; expected str, list or tuple; found: {sample_sets!r}"
-            )
-
```

Comment:
Handling type errors isn't required any more, because the `@check_types` decorator ensures that function parameters will conform to the type annotations.